source/nodejs/adaptivecards-designer/src/card-designer.ts

```diff
@@ -106,8 +110,12 @@ export class CardDesigner extends Designer.DesignContext {
                     '<div style="padding: 8px; display: flex; justify-content: center;">' +
                         '<div>The Card Structure isn\'t available in Preview mode.</div>' +
                     '</div>';
-            }
-            else {
+            } else if (this._isPayloadCardArray) {
+                this.treeViewToolbox.content.innerHTML =
+                    '<div style="padding: 8px; display: flex; justify-content: center;">' +
+                        '<div>The Card Structure isn\'t available for this payload.</div>' +
+                    '</div>';
```

Comment:
it's more of a pain, but rather than supplying content via `innerHTML` (which is considered a bad practice generally -- `""` is okay), we should be constructing the element through the DOM API... Something like:

```typescript
let toolboxDisabled = new HTMLElement("div");
toolboxDisabled.style.padding = "8px";
toolboxDisabled.style.display = "flex";
toolboxDisabled.style.justify-content = "center";
toolboxDisabled.textContent = "The card structure isn't available ...";
this._treeViewToolbox.content.appendChild(toolboxDisabled);
```

Maybe drop this in a helper method that lets you pass in the text to display :)